src/language-js/printer-estree.js

```diff
@@ -3679,31 +3679,38 @@ function printStatementSequence(path, options, print) {
 
 function printPropertyKey(path, options, print) {
   const node = path.getNode();
+  const parent = path.getParentNode();
   const key = node.key;
 
+  const objectHasStringProp = (
+    parent.properties ||
+    parent.body ||
+    parent.members
+  ).some(
+    prop =>
+      prop.key &&
```

Comment:
⚠️  This check may be hiding some bugs - need to remove it and investigate test failures.

Reply:
👍 False alarm, commenting this out causes no test failures. I'll leave it in to be a little defensive though.